### PR TITLE
Update README.md

### LINKED ISSUE #36: Update `README.md`
- getting started instructions

### PR DESCRIPTION
**Pull request by schemar**
Initial version of the readme for mosaic.js.

* Added all relevant badges
* Added same introduction as on mosaic-contracts
* Added installation section
* Moved facilitator readme into main readme
* Added preliminary instructions on how to deploy gateways
* Added mosaic-contracts as related work
* Added section on how to contribute

Note that I don't think the gateway deployment can be explained
concisely in the readme in the current state of the `develop` branch. We
can add that once it is feasible.

Fixes #36

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -1,105 +1,222 @@
...
+  )
+  .then(console.log)
+  .catch(console.log);
 ```
 
+#### Deploying Gateways
+
+Using `mosaic.js` to deploy gateways is still a work in progress and must be done manually for now:
+```js
+// Expecting Mosaic impart and an instance of `mosaic`:
```
Comment (hobofan) [inline README.MD]:
```suggestion
// Expecting Mosaic import and an instance of `mosaic`:
```

File: README.MD [hunk 2]
```diff
@@ -1,105 +1,222 @@
...
+const { EIP20Gateway } = Mosaic.Contracts;
+
+EIP20Gateway
+  .setup(
+    mosaic.origin.web3,
+    mosaic.auxiliary.web3,
+    gatewayConfig,
```
Comment (hobofan) [inline README.MD]: The contents of `gatewayConfig` and `coGatewayConfig` are probably the things that need the most explanation about the setup, and without that I'm not sure if the docs here provide much value.

Maybe add we should at least add a link to the typedef of the configs in the code until we have rendered & hosted docs we can link to?

**Review by hobofan — CHANGES_REQUESTED** [review]

File: README.MD (hunk 2 above)
Comment (schemar) [reply README.MD]: Yeah it is difficult. I had objects and short explanations, but it results in even more questions. 
Where do I get the addresses from? How do I deploy them if I need to?

Maybe we should remove the whole section from the readme. It is not ready, yet.

**Comment by deepesh-kn** [comment]
Should we include documentation on how to use the contract interacts?

File: README.MD [hunk 3]
```diff
@@ -1,105 +1,200 @@
...
 
-## Create configurations. 
-Make sure all deployer addresses, organization owner addresses and chainOwner have sufficient gas and are added to web3.eth.accounts.wallet of respective chains.
+Staking:
+```js
+const staker = '0x00000000000000000000stakeraccountaddress';
+const amount = '1000000000';
+const beneficiary = '0x000000000000000beneficiaryaccountaddress';
+const gasPrice = '1';
+const gasLimit = '1000000000';
```
Comment (deepesh-kn) [inline README.MD]: Please adjust the value of `amount`, `gasPrice` and `gasLimit` such that `amount > gasPrice * gasLimit`
This condition is not added in Facilitator.js but will fail in the contract.
Or maybe we should mention this in readme.

File: README.MD [hunk 4]
```diff
@@ -1,105 +1,244 @@
...
-    "deployer": "0x...", /* Address of key that will deploy the contract */
-  },
-  cogateway: {
-    "deployer": "0x...", /* Address of key that will deploy the contract */
-    "bounty": '1000000000000000000'
-  }
+const redeemer = '0x0000000000000000000000000redeemeraddress';
+const redeemAmount = '1000000000000';
+const beneficiary = '0x000000000000000beneficiaryaccountaddress';
+const gasPrice = '4;
```
Comment (deepesh-kn) [inline README.MD]:
```suggestion
const gasPrice = '4';
```

**Review by deepesh-kn — APPROVED** [review]
LGTM 👍 

Should this include any information about wrapping of OSTPrime for redeem ?

File: README.MD (hunk 4 above)
Comment (schemar) [reply README.MD]: Will do

**Comment by schemar** [comment]
I will add a note on wrapping and unwrapping

**Comment by schemar** [comment]
Add an extra section on OSTPrime

**Review by hobofan — APPROVED** [review]

**Comment by schemar** [comment]
Update to new Mosaic structure

File: README.MD [hunk 5]
```diff
@@ -1,105 +1,252 @@
...
+It composes heterogeneous blockchain systems into one another.
+Decentralized applications can use Mosaic to compute over a composed network of multiple blockchain systems in parallel.
 
-# Install Package
-```js
-npm install @openstfoundation/mosaic.js
-Mosaic = require('@openstfoundation/mosaic.js');
+Mosaic enables building scalable blockchain token economies through the bidirectional transposition of ERC20 tokens on one blockchain, the *origin* chain, and a utility token representation on another blockchain, the *auxiliary* chain.
+
+The protocol defines a set of actions that together perform atomic token transfers across two blockchains using gateway contracts. A gateway for a given EIP20 token is comprised of a `EIP20Gateway` contract on origin, a corresponding `EIP20CoGateway` contract on auxiliary, and and an ERC20 utility token contract on auxiliary that mints and burns utility tokens to atomically mirror tokens staked and unstaked on the origin chain.
```
Comment (deepesh-kn) [inline README.MD]:
```suggestion
The protocol defines a set of actions that together perform atomic token transfers across two blockchains using gateway contracts. A gateway for a given EIP20 token is comprised of a `EIP20Gateway` contract on origin, a corresponding `EIP20CoGateway` contract on auxiliary, and an EIP20 utility token contract on auxiliary that mints and burns utility tokens to atomically mirror tokens staked and unstaked on the origin chain.
```

File: README.MD [hunk 6]
```diff
@@ -1,105 +1,252 @@
...
+
+Atomicity is achieved using a 2-phase message passing architecture between the chains. Messages are declared on the source chain, and confirmed on the target chain with Patricia Merkle proofs once the source chain is finalized. Once messages are confirmed on the target chain, they can efficiently progressed with a hashlock.
```
Comment (deepesh-kn) [inline README.MD]:
```suggestion
Atomicity is achieved using a 2-phase message passing architecture between the chains. Messages are declared on the source chain, and confirmed on the target chain with Merkle Patricia proofs once the source chain is finalized. Once messages are confirmed on the target chain, they can efficiently progressed with a hashlock.
```

File: README.MD [hunk 7]
```diff
@@ -1,105 +1,252 @@
...
+Messages can also be reverted if they are not yet completed on the target chain.
```
Comment (deepesh-kn) [inline README.MD:18]: But we have not provided this support in mosaic.js yet.

File: README.MD [hunk 8]
```diff
@@ -1,105 +1,252 @@
...
+  .catch(console.log);
+```
+
+#### OSTPrime
+
+On the auxiliary chain OST is the base token.
+The gas price on auxiliary is paid in OST (like Ether pays for gas on Ethereum mainnet).
+To obtain the base token, you need to stake OST on origin.
+<!-- Add contract addresses for test nets -->
+On auxiliary, the `OSTPrime` contract will convert the EIP20 token to base tokens for the beneficiary.
```
Comment (deepesh-kn) [inline README.MD:214]: Should we provide a code snippet to show `wrap` and `unwrap`?

Comment (deepesh-kn) [reply README.MD:214]: If yes, I can help to provide the code snippet.

Comment (schemar) [reply README.MD:214]: Please do. Thank you! 🎉

File: README.MD (hunk 7 above)
Comment (schemar) [reply README.MD]:
```suggestion
Messages can also be reverted if they are not yet completed on the target chain.

> ⚠️ Reverting is not yet supported in `mosaic.js` and will be supported in a future version.
> In the meantime you can revert by accessing the contracts directly.
```

**Review by schemar — COMMENTED** [review]
@deepesh-kn please see

File: README.MD (hunk 8 above)
Comment (deepesh-kn) [reply README.MD:214]: Create OSTPrime object
```
const ostPrime = new Mosaic.ContractInteract.OSTPrime(
  mosaic.auxiliary.web3,
  mosaic.auxiliary.contractAddresses.OSTPrime,
);

```
Wrap
```
const amountToWrap = '1000';
const fromAddress = '0x00000000000000000000000000000fromaddress';
let txOptions = {
  value: amountToWrap,
  from: fromAddress,
};

ostPrime.wrap(txOptionsAuxiliaryWithValue).then(console.log).catch(console.log);
```

Unwrap
```
const amountToUnwrap = '1000';
txOptions = {
  from: fromAddress,
};

ostPrime.unwrap(amountToUnwrap, txOptions).then(console.log).catch(console.log);
```

**Comment by hobofan** [comment]
🚀 🚀 🚀

File: README.MD [hunk 9]
```diff
@@ -1,105 +1,291 @@
...
+    gasPrice,
+    gasLimit,
+    hashLock,
+    txOptions,
+  )
+  .then(console.log)
+  .catch(console.log);
 ```
 
+Progressing the stake after the state root has been transferred:
```
Comment (0xsarvesh) [inline README.MD:91]: should we add any reference, how this is done?

Comment (0xsarvesh) [reply README.MD:91]: https://github.com/OpenSTFoundation/mosaic-anchor.js

**Review by 0xsarvesh — APPROVED** [review]
LGTM 🐎  
Just one question/suggestion.

**Review by benjaminbollen — APPROVED** [review]
LGTM
giphy-3

File: README.MD (hunk 9 above)
Comment (benjaminbollen) [reply README.MD:91]: we need a more concerted effort to build documentation immediately after beta.1 release

**Pull request merged**